pkg/networkmanager/interfaces.js

```diff
@@ -3317,31 +3314,45 @@ function set_slave(model, master_connection, master_settings, slave_type,
         if (!master_iface)
             return false;
 
-        if (!main_connection) {
-            if (!iface.Device)
-                return false;
-
-            return iface.Device.activate_with_settings({ connection:
-                                                         { autoconnect: true,
-                                                           interface_name: iface.Name,
-                                                           slave_type: slave_type,
-                                                           master: master_iface
-                                                         }
-                                                       });
-        } else if (cs.master != master_settings.connection.uuid &&
-                   cs.master != master_settings.connection.id &&
-                   cs.master != master_iface) {
-            cs.slave_type = slave_type;
-            cs.master = master_iface;
-            main_connection.Settings.connection.autoconnect = true;
-            delete main_connection.Settings.ipv4;
-            delete main_connection.Settings.ipv6;
-            delete main_connection.Settings.team_port;
-            delete main_connection.Settings.bridge_port;
-            return main_connection.apply_settings(main_connection.Settings).then(function () {
-                return main_connection.activate(iface.Device, null);
-            });
+        var slave_settings;
+        if (main_connection) {
+            slave_settings = main_connection.Settings;
+
+            if (slave_settings.connection.master == master_settings.connection.uuid ||
+                slave_settings.connection.master == master_settings.connection.id ||
+                slave_settings.connection.master == master_iface)
+                return cockpit.resolve();
+
+            slave_settings.connection.slave_type = slave_type;
+            slave_settings.connection.master = master_iface;
+            slave_settings.connection.autoconnect = true;
```

Comment:
> I thought the idea of this PR was that the master sets autoconnect-slaves = 1 for these, instead of using autoconnect on the slaves?

Good point.  Setting `autoconnect = true` on the slaves was an explicit request by the users.  Setting `autoconnect_slaves = 1` on the master seems to be necessary to actually activate the slaves with the master.

I don't know what's going on in detail, tbh.